ext/totals.py

```diff
@@ -109,11 +109,8 @@ def calculate_totals(input_stream):
         else:
             report_end = datetime.datetime.strptime(j[-1]["end"], DATEFORMAT).replace(tzinfo=from_zone)
     else:
-        if report_end_utc is not None:
-            j[-1]["end"] = report_end_utc.strftime(DATEFORMAT)
-        else:
-            j[-1]["end"] = datetime.datetime.now(tz=from_zone).strftime(DATEFORMAT)
```

Comment:
You are right that overwriting the end time of an open interval with the report end is wrong, but only if the report end time is *after* the current time.

If one would call `timew totals :yesterday` with an interval started yesterday, your fix would also display the time of today.

So I think we should take the minimum of _now_ and _report.end_ instead.

Reply:
that makes sense, updated.